### PR TITLE
Jj1.4featuresv2

### PR DESCRIPTION
**Pull request by teojunjie**
Mainly fuzzy wuzzy findmachine

### CODE REVIEW

**Comment by Psyf** [comment]
Did you approve fuzzywuzzy from akshay?

**Comment by teojunjie** [comment]
> Did you approve fuzzywuzzy from akshay?

Yes but he say must do some licensing shit

**Comment by teojunjie** [comment]
https://opensource.stackexchange.com/questions/1640/if-im-using-a-gpl-3-library-in-my-project-can-i-license-my-project-under-mit-l

**Comment by teojunjie** [comment]
@Psyf ready for review

**Comment by teojunjie** [comment]
@Psyf Completed the auto add function to other machines instead of just one . Do review the code, only coveralls failing

**Comment by Psyf** [comment]
Let me wait till approval and licensing stuff, otherwise might be messy to take out.

**Comment by teojunjie** [comment]
image

**Comment by teojunjie** [comment]
I dont understand how to the do the licensing

**Comment by teojunjie** [comment]
https://github.com/xdrop/fuzzywuzzy

**Comment by teojunjie** [comment]
@Psyf Do you think this will be adequate if i put at the files where i use the code?

This file is part of Foobar.

Foobar is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

Foobar is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with Foobar.  If not, see <http://www.gnu.org/licenses/>.

**Comment by teojunjie** [comment]
Or maybe i will use this instead?

Credits
This application uses Open Source components. You can find the source code of their open source projects along with license information below. We acknowledge and are grateful to these developers for their contributions to open source.

Project: Express-Combo https://github.com/yahoo/express... 
Copyright 2013 Yahoo! Inc. All right reserved.
License (3-Clause BSD) https://github.com/yahoo/express...

Project: MNColorKit  https://github.com/madninja/MNCo...
Copyright (c) 2013 Marc Nijdam (marc@imadjine.com)
License (MIT) https://github.com/madninja/MNCo...

Project: YUI3  https://github.com/yui/yui3
Copyright (c) 2013, Yahoo! Inc. All right reserved.
License (3-Clause BSD) https://github.com/yui/yui3/blob...

**Comment by teojunjie** [comment]
Ok @Psyf this is the draft for credits i m not sure if this appropriate enough. I will post on the forum to see if this is ok

Credits
This application uses Open Source components. You can find the source code of their open source projects along with license information below. We acknowledge and are grateful to these developers for their contributions to open source.

Project: JavaWuzzy https://github.com/xdrop/fuzzywuzzy
Copyright (C) 2007 Free Software Foundation, Inc. <http://fsf.org/>
GNU General Public License v3.0 https://github.com/xdrop/fuzzywuzzy/blob/master/LICENSE

**Comment by Psyf** [comment]
So follow up is apparently the licensing wasn't enough and we stopped using fuzzy wuzzy?

**Comment by teojunjie** [comment]
@Psyf Removed fuzzywuzzy, can be merged now.

**Comment by Psyf** [comment]
Can we use F10-03's with their permission?

**Comment by teojunjie** [comment]
> So follow up is apparently the licensing wasn't enough and we stopped using fuzzy wuzzy?

Yes

**Comment by teojunjie** [comment]
> Can we use F10-03's with their permission?

Emm i can do the algorithm my own ah, but it will take time

File: src/main/java/seedu/address/logic/commands/machine/ManageMachineCommand.java [hunk 1]
```diff
@@ -104,11 +104,7 @@ public CommandResult execute(Model model, CommandHistory history) throws Command
                         return new CommandResult(String.format(MESSAGE_MACHINE_NOT_FLUSHED, machineToManage));
                     }
                 } else if (flushAutoOption.equals(OPTION_FLUSH_AUTO)) {
-                    Machine mostFreeMachine = model.getMostFreeMachine();
-                    if (mostFreeMachine.equals(machineToManage)) {
-                        throw new CommandException(MESSAGE_NO_MORE_MACHINES);
-                    }
-                    model.autoMoveJobs(machineToManage, mostFreeMachine);
+                    model.autoMoveJobs(machineToManage);
```
Comment (Psyf) [inline src/main/java/seedu/address/logic/commands/machine/ManageMachineCommand.java]: Can you rename the function to autoMoveJobsDuringFlush() or sth? Getting confusing now.

Comment (teojunjie) [reply src/main/java/seedu/address/logic/commands/machine/ManageMachineCommand.java]: Done

**Pull request merged**